www/utils/supabase/admin.ts

```diff
@@ -21,6 +22,71 @@ const supabaseAdmin = createClient<Database>(
   process.env.SUPABASE_SERVICE_ROLE_KEY || ''
 );
 
+// Trial membership
+const createOrRetrieveFreeTrialSubscription = async (userId: string) => {
+  // Check for existing trial subscription
+  const { data: existingSub, error: subError } = await supabaseAdmin
+    .from('subscriptions')
+    .select('*')
+    .eq('user_id', userId)
+    .eq('status', 'trialing')
+    .maybeSingle();
+
+  if (subError) throw new Error(`Subscription lookup failed: ${subError.message}`);
+
+  // If trial subscription exists, return it
+  if (existingSub) return existingSub;
+
+  // Create a new trial subscription
+  const subscriptionData: TablesInsert<'subscriptions'> = {
+    id: `free_trial_${userId}`,
+    user_id: userId,
+    status: 'trialing',
+    metadata: { freeMessages: FREE_MESSAGE_LIMIT },
+    price_id: null, // or your free tier price ID if you have one
+    quantity: 1,
+    cancel_at_period_end: false,
+    created: new Date().toISOString(),
+    current_period_start: new Date().toISOString(),
+    current_period_end: new Date(Date.now() + 365 * 24 * 60 * 60 * 1000).toISOString().replace('T', ' ').replace('Z', ''), // 1 year
```

Comment:
Need to think through if it makes sense to send an end date for a year or just have no end date